terminado/tests/basic_test.py

```diff
@@ -5,42 +5,223 @@
+
+        return tornado.web.Application([
+                    (r"/new",         NewTerminalHandler),
+                    (r"/named/(\w+)", TermSocket, {'term_manager': named_tm}),
+                    (r"/single",      TermSocket, {'term_manager': single_tm}),
+                    (r"/unique",      TermSocket, {'term_manager': unique_tm})
+                ], debug=True)
+
+    test_urls = ('/named/term1', '/unique', '/single')
+
+class CommonTests(TermTestCase):
     @tornado.testing.gen_test
     def test_basic(self):
-        ws = yield self.ws_connect("/websocket/term1")
-        response = yield ws.read_message()              # Setup message
-        j = json.loads(response)
-        self.assertEqual(j, ['setup', {}])
-
-        response = yield ws.read_message()              # Command prompt
-        j = json.loads(response)
-        self.assertEqual(j[0], 'stdout')
-        ws.close()
+        for url in self.test_urls:
+            tm = yield self.get_term_client(url)
+            response = yield tm.read_msg()                    
+            self.assertEqual(response, ['setup', {}])
+
+            # Check for initial shell prompt
+            response = yield tm.read_msg()
+            self.assertEqual(response[0], 'stdout')
+            self.assertGreater(len(response[1]), 0)
+            tm.close()
+
+    @tornado.testing.gen_test
+    def test_basic_command(self):
+        for url in self.test_urls:
+            tm = yield self.get_term_client('/named/foo')
+            yield tm.read_all_msg()
+            tm.write_stdin("whoami\r")
+            (stdout, other) = yield tm.read_stdout()
+            self.assertEqual(stdout[:6], "whoami")
+            self.assertEqual(other, [])
+            tm.close()
+
+class NamedTermTests(TermTestCase):
+    @tornado.testing.gen_test
+    def test_no_name(self):
+        with self.assertRaises(HTTPError) as context:
+            tm = yield self.get_term_client('/named/')
+            yield tm.read_msg()
+
+        # Not found
+        self.assertEqual(context.exception.code, 404)
+        self.assertEqual(context.exception.response.code, 404)
+
+    def test_new(self):
+        response = self.fetch("/new", follow_redirects=False)
+        self.assertEqual(response.code, 302)
+        url = response.headers["Location"]
+
+        # Check that the given terminal name works
+        tm = self.get_term_client(url)
+        tm.add_done_callback(self.stop)
+        self.wait()
+
+    @tornado.testing.gen_test
+    def test_namespace(self):
+        names = ["/named/1"]*2 + ["/named/2"]*2
+        tms = yield self.get_term_clients(names)
+        pids = yield self.get_pids(tms)
+
+        self.assertEqual(pids[0], pids[1])
+        self.assertEqual(pids[2], pids[3])
+        self.assertNotEqual(pids[0], pids[3])
+
+    @tornado.testing.gen_test
+    def test_max_terminals(self):
+        urls = ["/named/%d" % i for i in range(MAX_TERMS+1)]

```

Comment:
I just mean that since this test doesn't currently create the (N+1)th terminal, it's not really checking anything that isn't checked elsewhere, so we may as well leave it out until we work out how to test what it aims to test.